jgrapht-core/src/main/java/org/jgrapht/alg/cycle/ChordalityInspector.java

```diff
@@ -0,0 +1,211 @@
+package org.jgrapht.alg.cycle;
+
+import org.jgrapht.Graph;
+import org.jgrapht.Graphs;
+import org.jgrapht.graph.AsUndirectedGraph;
+import org.jgrapht.traverse.LexicographicalBfsIterator;
+import org.jgrapht.traverse.MaximumCardinalityIterator;
+
+import java.util.*;
+
+/**
+ * Implementation of the lexicographical breadth-first search algorithm for chordal graph recognition.
+ * <p>
+ * A <a href="https://en.wikipedia.org/wiki/Chordal_graph">chordal graph</a> is one in which all cycles of four or more vertices have a chord, which is an edge that
+ * is not part of the cycle but connects two vertices of the cycle.
+ * <p>
+ * A graph is chordal iff its the vertices can be arranged into a perfect elimination order. Perfect elimination
+ * order isn't unique. Both maximum cardinality search and lexicographical breadth-first search produces this order.
+ * As a result, chordality inspection can be performed in two way. Here it is done by default via MCS,
+ * because it outperforms LexBFS by a constant factor in practice. Nevertheless, lexicographical breadth-first
```

Comment:
I would refrain from stating such facts without the proper reference. It there some particular paper that has done the experimental comparison and state that? Have you done some experiments and published the results somewhere?